tools/make.py

```diff
@@ -173,6 +186,11 @@
                       default=False,
                       help="Link with mbed test library")
 
+    parser.add_argument("--metadata",
```

Comment:
yesh metadata might not be the best name for this. I was also thinking do we need this parameter at all - could we generate it just always - just like we do with test_spec..?